index.js

```diff
@@ -0,0 +1,138 @@
+module.exports = {
+  env: {
+    node: true,
+    es2020: true,
+    "jest/globals": true,
+  },
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:prettier/recommended",
+    "prettier/@typescript-eslint",
+    "plugin:jest/recommended",
+    "plugin:json/recommended",
+  ],
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    project: "tsconfig.json",
+    sourceType: "module",
+    ecmaFeatures: {
+      jsx: true,
+    },
+  },
+  plugins: [
+    "@typescript-eslint",
+    "@typescript-eslint/tslint",
+    "jest",
+    "prettier",
+    "import",
+    "json",
+  ],
+  overrides: [
+    {
+      files: ["*.js"],
+      rules: {
+        "@typescript-eslint/no-var-requires": "off",
+        "@typescript-eslint/explicit-function-return-type": "off",
+      },
+    },
+    {
+      files: [
+        "babel.config.js",
+        "jest.config.js",
+        "release.config.js",
+        ".eslintrc.js",
+        "*.json",
+      ],
+      parser: "esprima",
+    },
+  ],
+  rules: {
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      {
+        varsIgnorePattern: "^_",
+        argsIgnorePattern: "^_",
+        ignoreRestSiblings: true,
+      },
+    ],
+    "@typescript-eslint/camelcase": "off",
+    "@typescript-eslint/naming-convention": [
+      "error",
+      {
+        selector: "default",
+        format: ["camelCase"],
+      },
+      {
+        selector: "memberLike",
+        format: ["camelCase", "PascalCase"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "variableLike",
+        format: ["camelCase", "PascalCase"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "property",
+        format: ["camelCase", "snake_case", "PascalCase"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "property",
+        format: null,
+        filter: {
+          regex: "^__html$",
+          match: true,
+        },
+      },
+      {
+        selector: "method",
+        format: null,
+        filter: {
+          regex: "^UNSAFE_",
+          match: true,
+        },
+      },
+      {
+        selector: "typeLike",
+        format: ["PascalCase"],
+      },
```

Comment:
do we really have such a custom setup that we need to specify all these?

Reply:
Take a look here, that's my summary of what we cover: https://github.com/carforyou/carforyou-eslint-rules-pkg/pull/1#discussion_r427186803

This rule is not enabled by default, `camelcase` is. But:
1) `camelcase` is deprecated in favour of this one
2) this one actually gives us the flexibility to cover all of our cases and some control if we want to change naming conventions in the future.